docs/index.md

```diff
@@ -86,24 +86,87 @@ connection "terraform" {
+- `../*.tf` matches all Terraform configuration files in the CWD's parent directory.
+- `steampipe*.tf` matches all Terraform configuration files starting with "steampipe" in the CWD.
+- `/path/to/dir/*.tf` matches all Terraform configuration files in a specific directory. For example:
+  - `~/*.tf` matches all Terraform configuration files in the home directory.
+  - `~/**/*.tf` matches all Terraform configuration files recursively in the home directory.
+- `/path/to/dir/main.tf` matches a specific file.
+
+```hcl
+connection "terraform" {
+  plugin = "terraform"
+
+  paths = [ "*.tf", "~/*.tf", "/path/to/dir/main.tf" ]
+}
+```
+
+**NOTE:** If paths includes `*`, all files (including non-Terraform configuration files) in the CWD will be matched, which may cause errors if incompatible file types exist.
+
+#### Configuring GitHub URLs
+
+You can define a list of URL as input to search for terraform files from a variety of protocols. For example:
+
+- `github.com/turbot/polygoat//*.tf` matches all top-level Terraform configuration files in the specified github repository.
+- `github.com/turbot/polygoat//**/*tf` matches all Terraform configuration files in the specified github repository and all sub-directories.
+- `github.com/turbot/polygoat?ref=fix_7677//**/*tf` matches all Terraform configuration files in the specific tag of github repository.
+- `git::https://github.com/turbot/steampipe-plugin-alicloud.git//alicloud-test/tests/alicloud_account//*.tf` matches all Terraform configuration files in the given HTTP URL using the Git protocol.
+
+If you want to download only a specific subdirectory from a downloaded directory, you can specify a subdirectory after a double-slash (`//`).
+
+- `github.com/turbot/polygoat//testing_frameworks/steampipe_mod_benchmark//*.tf` matches all Terraform configuration files in the specific folder of a github repo.
+
+```hcl
+connection "terraform" {
+  plugin = "terraform"
+
+  paths = [ "github.com/turbot/polygoat//*.tf", "github.com/turbot/polygoat//testing_frameworks/steampipe_mod_benchmark//*.tf", "git::https://github.com/turbot/steampipe-plugin-alicloud.git//alicloud-test/tests/alicloud_account//*.tf" ]
 }
 ```
 
-- `paths` - A list of directory paths to search for Terraform files. Paths are resolved relative to the current working directory. Paths may [include wildcards](https://pkg.go.dev/path/filepath#Match) and also support `**` for recursive matching. Defaults to the current working directory.
+#### Configuring S3 URLs
+
+You can also pass a S3 bucket URL to search all Terraform configuration files stored in the specified S3 bucket. For example:
+
+- `s3::https://s3.amazonaws.com/bucket/terraform_examples//**/*.tf` matches all the Terraform configuration files recursively.
+
+```hcl
+connection "terraform" {
+  plugin = "terraform"
+
+  paths = [ "s3::https://s3.amazonaws.com/bucket/terraform_examples//**/*.tf" ]
+}
+```
 
```

Comment:
I think we should also support GCP buckets, as go-getter seems to support it (https://github.com/hashicorp/go-getter#gcs-gcs).

I saw the issue opened by another user, that we already commented on as well, https://github.com/hashicorp/go-getter/issues/358, but I'm wondering if we could fork, make the changes from `gcs` to `gs`, submit back a PR, and use the fork in the meantime.